azurerm/resource_arm_app_insights.go

```diff
@@ -0,0 +1,138 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/appinsights"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAppInsights() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppInsightsCreateOrUpdate,
+		Read:   resourceArmAppInsightsRead,
+		Update: resourceArmAppInsightsCreateOrUpdate,
+		Delete: resourceArmAppInsightsDelete,
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"kind": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": locationSchema(),
+			"tags":     tagsSchema(),
+			"instrumentation_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAppInsightsCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	AppInsightsClient := client.appInsightsClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM App Insights creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	insightsType := "microsoft.insights/components"
+	kind := d.Get("kind").(string)
+	location := d.Get("location").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	requestSource := "IbizaAIExtension"
```

Comment:
From what I can see, this can only be set to:
```
// RequestSource enumerates the values for request source.
type RequestSource string

const (
	// Rest specifies the rest state for request source.
	Rest RequestSource = "rest"
)
```